spec/services/google/client_service.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Google
+  class ClientService
+    USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) '\
+                 'AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.77 Safari/537.36'
+
+    BASE_SEARCH_URL = 'https://www.google.com/search'
+
+    def initialize(keyword:, lang: 'en')
+      @escaped_keyword = CGI.escape(keyword)
+      @uri = URI("#{BASE_SEARCH_URL}?q=#{@escaped_keyword}&hl=#{lang}&gl=#{lang}")
+    end
+
+    def call
+      result = HTTParty.get(@uri, { headers: { 'User-Agent' => USER_AGENT } })
+
+      return false unless valid_result? result
+
+      result
+    rescue HTTParty::Error, Timeout::Error, SocketError => e
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mosharaf13/fleet-searcher-ruby/blob/9c2993cee1849a752110f1b742a6a7e32a4b619e/spec/services/google/client_service.rb#L21"></span>Has the variable name 'e'</td>
    </tr>
  </tbody>
</table>